utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -174,8 +174,12 @@ def install_toolchain_path(self, host_target):
 
         install_destdir = self.args.install_destdir
         if self.args.cross_compile_hosts:
-            build_root = os.path.dirname(self.build_dir)
-            install_destdir = '%s/intermediate-install/%s' % (build_root, host_target)
+            if host_target == self.args.host_target:
```

Comment:
I use this check throughout this pull to differentiate between cross-compiling for macOS versus non-Darwin targets, because only the latter sets a different `host_target` above.